content/hello-world-program.md

```diff
@@ -0,0 +1,295 @@
+  program_id: &Pubkey,
+  accounts: &[AccountInfo],
+  instruction_data: &[u8]
+) -> ProgramResult{
+  msg!("Hello world!");
+
+  Ok(())
+}
+```
+
+All together, the hello world program will look like this:
+
+```rust
+use solana_program::{
+  account_info::AccountInfo,
+  entrypoint,
+  entrypoint::ProgramResult,
+  pubkey::Pubkey,
+  msg
+};
+
+entrypoint!(process_instruction);
+
+pub fn process_instruction(
+  program_id: &Pubkey,
+  accounts: &[AccountInfo],
+  instruction_data: &[u8]
+) -> ProgramResult{
+  msg!("Hello, world!");
+
+  Ok(())
+}
+```
+
+Now lets build and deploy our program using Solana Playground.
+
+![Gif Solana Playground Build and Deploy](../assets/hello-world-build-deploy.gif)
+
+Finally, lets invoke our program from the client side. Download the code [here](https://github.com/ZYJLiu/solana-hello-world-client).
+
+The focus of this lesson is to build our Solana program, so we’ve gone ahead and provided the client code to invoke our hello world program. The code provided includes a `sayHello` helper function that builds and submits our transaction. We then call `sayHello` in the main function and print a Solana Explorer URL to view our transaction details in the browser. 
+
+In the `index.ts` file you should see the following: 
+
+```tsx
+import web3 = require("@solana/web3.js");
+import Dotenv from "dotenv";
+Dotenv.config();
+
+let programId = new web3.PublicKey("<YOUR_PROGRAM_ID>");
+
+let payer = initializeKeypair();
+let connection = new web3.Connection(web3.clusterApiUrl("devnet"));
+
+async function main() {
+  await connection.requestAirdrop(payer.publicKey, web3.LAMPORTS_PER_SOL * 1);
+
+  const transactionSignature = await sayHello();
+
+  console.log(
+    `Transaction: https://explorer.solana.com/tx/${transactionSignature}?cluster=custom&customUrl=http%3A%2F%2Flocalhost%3A8899`
+  );
+}
+
+main()
+  .then(() => {
+    console.log("Finished successfully");
+  })
+  .catch((error) => {
+    console.error(error);
+  });
+
+export async function sayHello(): Promise<web3.TransactionSignature> {
+  const transaction = new web3.Transaction();
+
+  const instruction = new web3.TransactionInstruction({
+    keys: [],
+    programId,
+    data: Buffer.alloc(0),
+  });
+
+  transaction.add(instruction);
+
+  const transactionSignature = await web3.sendAndConfirmTransaction(
+    connection,
+    transaction,
+    [payer]
+  );
+
+  return transactionSignature;
+}
+
+function initializeKeypair(): web3.Keypair {
+  const secret = JSON.parse(process.env.PRIVATE_KEY ?? "") as number[];
+  const secretKey = Uint8Array.from(secret);
+  const keypairFromSecretKey = web3.Keypair.fromSecretKey(secretKey);
+  return keypairFromSecretKey;
+}
+```
+
+Update the `programId` with the your program id from Solana Playground.
+
+```rust
+let programId = new web3.PublicKey("<YOUR_PROGRAM_ID>");
+```
+
+![Gif Solana Playground Program Id](../assets/hello-world-program-id.gif)
+
+Next, run `npm start` in the terminal.
```

Comment:
@ZYJLiu I think we could lose "in the terminal"?

Also, next line for the img descr. we could just say "Gif npm start"